0000-py-version-support.md

```diff
@@ -0,0 +1,244 @@
+----------------------------
+Another motivating factor to make this policy change is our shift to including Python in our
+packages. For the Sodium release we will be using salt-bin (https://github.com/saltstack/salt-bin) to
+package for our Fully Supported Platforms.
+(http://get.saltstack.com/rs/304-PHQ-615/images/SaltStack-Supported-Operating-Systems.pdf)
+This will allow us to easily continue supporting operating systems that include End of Life Python
+versions by default by providing the Python binary in the Salt package included in a Salt release.
+
+Security and Maintenance Concerns
+---------------------------------
+The last motivating factor surrounds security and maintenance concerns. Continuing to support an End
+of Life Python version adds additional maintenance around testing and development for each version.
+Supporting these versions also puts users at a security risk, since no more security fixes will be
+submited and released with that Python version if the operating system does not choose to update
+it themselves.
+
+# Design
+[design]: #detailed-design
+
+The design goal of this SEP is to define a clear policy around Salt's Python version support.
+This new policy defines only supporting major Python versions until they reach End of Life. You can see
+the current status of supported Python branches here https://devguide.Python.org/#status-of-Python-branches
+with the associated End Of Life date for each branch. You can also see details about Python versions that
+have already become End Of Life here: https://devguide.Python.org/devcycle/#end-of-life-branches.
+
+What if a Python version drops support relatively close to a Salt Release?
+--------------------------------------------------------------------------
+We may drop support for a Python version if the End of Life date lands relatively close to a Salt release.
+For example if a Python version is End of Life a couple weeks before a Salt release, support for that
+Python version may drop in that Salt version. This will need to be clearly communicated beforehand
+with the community.
+
+How will this impact the Sodium release?
+----------------------------------------
+Because Python 3.4 became End Of Life on 2019-03-18, we will only support Python versions >= 3.5
+
+How does this impact all supported Operating Systems
+----------------------------------------------------
+The details below will include the impact for our supported operating systmes included here
+and some future operating systems we will be supporting:
+http://get.saltstack.com/rs/304-PHQ-615/images/SaltStack-Supported-Operating-Systems.pdf
+
+
+AIX/Solaris
+-----------
+AIX/Solaris are Enterprise packages and will use salt-bin for their packaging mechanism,
+so they will include the correct Python version.
+
+Windows
+-------
+Python does not come pre-installed on Windows operating systems. We have always included
+the Python binary in the package and will include to do that going forward. If someone
```

Comment:
... and will continue to do that ...

Reply:
Though, it is my understanding that we will be using pop-build even on Windows, though the effect is still the same, that we are bundling python

Reply:
yes, i wanted to leave the details for salt-bin/pop-build for a different SEP as this is just focused on python version support. The important thing to note here is that Python is included.